### PR TITLE
Variable Support Continuation

### PR DESCRIPTION
**Pull request by hborawski**
addresses review comment on #2 

I'm looking to add function declaration to this project so i can define macros with the AST, and variable use would be a prerequisite

### CODE REVIEW

File: lib/starlark_compiler/build_file.rb [hunk 1]
```diff
@@ -36,10 +41,15 @@ def to_starlark
       loads = @loads
               .sort_by { |k, _| k }
               .map { |f, fn| AST.build { function_call('load', f, *fn.sort) } }
+      variable_assignments = @variable_assignments
+                             .sort_by { |k, _| k }
```
Comment (segiddins) [inline lib/starlark_compiler/build_file.rb]: I don't think sorting is safe, as they could be order dependent

File: lib/starlark_compiler/ast.rb [hunk 2]
```diff
@@ -101,13 +101,38 @@ def initialize(str)
       end
     end
 
+    class VariableReference < Node
+      attr_reader :var
+      def initialize(var)
+        unless var.is_a?(::String)
+          raise ''"
+Only string type is allowed as a variable reference, got #{var.class}
```
Comment (segiddins) [inline lib/starlark_compiler/ast.rb]: Please put the string on a single line

File: lib/starlark_compiler/ast.rb [hunk 3]
```diff
@@ -101,13 +101,38 @@ def initialize(str)
...
       def initialize(elements)
         @elements = elements.map(&method(:node))
       end
     end
 
+    class VariableAssignment < Node
+      attr_reader :name, :var
+      def initialize(name, var)
+        @name = name
+        if [VariableAssignment].include?(var.class)
```
Comment (segiddins) [inline lib/starlark_compiler/ast.rb]: Potentially worth marking nodes as statements vs expressions, and only allow expressions on the RHS?

Comment (hborawski) [reply lib/starlark_compiler/ast.rb]: I think that makes more sense long term

I don't write much ruby, so please let me know if theres a better way to implement this than i did (there probably is)

File: lib/starlark_compiler/ast.rb (hunk 2 above)
Comment (hborawski) [reply lib/starlark_compiler/ast.rb]: Looks like this was because of rubocop for line length, i reworked the message

**Comment by hborawski** [comment]
@segiddins Anything else needed to merge this?

**Comment by segiddins** [comment]
Released in 0.4.0

**Pull request merged**